src/relayserver/penalizer/PenalizerService.ts

```diff
@@ -140,15 +176,82 @@ export class PenalizerService {
       throw Error(`Failed to get transaction ${minedTransactionData.hash} from node`)
     }
     const minedTxBuffers = createWeb3Transaction(minedTx, rawTxOptions)
-    const method = this.getPenalizeRepeatedNonceMethod(minedTxBuffers, requestTx)
+    const randomValue = randomInRange(0, 0xffffffff).toString(16)
+    const penalizationArguments = this.getPenalizeRepeatedNonceArguments(minedTxBuffers, requestTx, randomValue)
```

Comment:
I really don't understand why you use getXXarguments and getMethod(XX):
instead, keep the method object itself in the delayedPenalization.

Reply:
I hate the concept of 'method object', I think it is best to avoid using it if there is a simple way to use a plain JavaScript object.

Reply:
You have the "object method" so much, that you have to create helper method for the params of each solidity function, and another switch/case helper method to select the right function....
I don't think its a good tradeoff.